app/repositories/application/GeneralApplicationRepository.scala

```diff
@@ -372,12 +372,19 @@ class GeneralApplicationMongoRepository(timeZoneService: TimeZoneService,
 
   override def findTestForNotification(notificationType: NotificationTestType): Future[Option[TestResultNotification]] = {
     val query = Try{ notificationType match {
-      case s: SuccessTestType => {
+      case s: SuccessTestType if s.applicationRoutes.isEmpty => {
```

Comment:
Not required. This application route filter will just make sure the notification is sent only for specific application types.